src/main/webapp/workspace/execution.js

```diff
@@ -28,24 +38,24 @@ function executeCode() {
   getToken().then(tok => {
     fetch(`/workspace/queueExecution?workspaceID=${getParam("workspaceID")}&idToken=${tok}`)
       .then(resp => resp.text()).then(execID => {
-        getFirebaseRef().child("executions").child(execID).on("value", snap => {
+        seenFirstOutput = false;
+        term.write('\x1bc'); // clear terminal
+
+        getFirebaseRef().child("executions").child(execID).on("child_added", snap => {
           if (snap.val() !== null) {
-            const pre = document.createElement("pre");
-            pre.innerText = snap.val();
-
-            const outputContainer = document.getElementById("output-container");
-            outputContainer.innerHTML = "";
-            outputContainer.appendChild(pre);
-
-            executeButton.disabled = false;
-            executeButton.classList.remove("download-in-progress");
-            getFirebaseRef().child("executions").child(execID).off("value");
-            
-            if (!outputVisible) {
-              toggleOutput();
-            }        
-
-            outputContainer.scrollTop = outputContainer.scrollHeight;
+            if (typeof snap.val() === 'string' || snap.val() instanceof String) {
```

Comment:
what is the difference between these two checks?

Reply:
Javascript has two types of strings one is a primitive (first check) and the other is an object wrapper (second check).